zipkin-server/server-starter/src/main/java/zipkin/server/config/ApplicationConfigLoader.java

```diff
@@ -119,7 +119,12 @@ private void replacePropertyAndLog(final String propertyName, final Object prope
     private Object convertValueString(String valueString) {
         try {
             Object replaceValue = yaml.load(valueString);
-            if (replaceValue instanceof String || replaceValue instanceof Integer || replaceValue instanceof Long || replaceValue instanceof Boolean || replaceValue instanceof ArrayList) {
+            if (replaceValue instanceof String ||
```

Comment:
I am wondering, why do we need to copy `ApplicationConfigLoader`? Is it possible to use `ZipkinServerBootstrap` calling the existing codes from SkyWalking starter codes?

Reply:
It follows these two reasons:
1. If further customization of configuration files is needed later on, this file allows for easy re-customization.
2. If based on the `server-starter` project, which includes all module projects, we need to remove dependencies sequentially. When new modules are introduced in Skywalking OAP and synchronized to Zipkin later, we still need to remove unrelated dependencies.

Reply:
> If further customization of configuration files is needed later on, this file allows for easy re-customization.

The biggest concern is duplicate `ApplicationConfigLoader`. It has complex logic and very general. Should we consider moving this class from OAP starter to core and make it available easily here?

Reply:
This means that we need to put all the application startup configuration readers in the core. I feel like this might cause some confusion? Because the core doesn't need to read configuration files.

Reply:
OK. Let's keep the duplicate codes for now. Personally, I don't like we copied the same codes.